iphone/cli/commands/_build.js

```diff
@@ -4409,6 +4411,71 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
 			writeAssetContentsFile.call(this, path.join(launchImageDir, 'Contents.json'), launchImageSet);
 		},
 
+		function createAssetImageSets() {
+			this.logger.info(__('Creating assets image set'));
+			var assetCatalog = path.join(this.buildDir, 'Assets.xcassets'),
+				imageSets = {},
+				imageNameRegExp = /([\w\-\.\/]+)(@\dx)?(~iphone|~ipad)?\./; //[imageSetName][@Nx][~device]
+
+			var sha = crypto.createHash('sha1');
+
+			function sha1(value) {
+				var sha = crypto.createHash('sha1');
+				sha.update(value);
+				return sha.digest('hex');
+			};
+
+			Object.keys(imageAssets).forEach(function (file) {
+				var imageName = imageAssets[file].name,
+					imageExt = imageAssets[file].ext,
+					match = file.match(imageNameRegExp),
+					imageSetName,
+					imageSetNameSHA,
+					imageSetRelPath,
+					imageSetFolder;
+
+				if (match) {
+					imageSetName = match[1];
+					imageSetNameSHA = sha1(imageSetName + '.' + imageExt);
+					imageSetRelPath = imageSetNameSHA + '.imageset';
+					imageSetAbsPath = path.join(assetCatalog, imageSetRelPath);
+
+					// update image file's destination
+					imageAssets[file].dest = path.join(imageSetAbsPath, imageName + '.' + imageExt);
+
+					if (!(imageSetRelPath in imageSets)) {
```

Comment:
Generally the `in` operator is frowned down upon. You can generally say `!imageSets[imageSetRelPath]`, but sometimes you need to do `!imageSets.hasOwnProperty(imageSetRelPath)`.